crates/engine/tree/src/tree/mod.rs

```diff
@@ -282,7 +282,7 @@ impl TreeState {
         }
 
         // remove trie updates that are below the finalized block
-        self.persisted_trie_updates.retain(|_, (block_num, _)| *block_num < finalized_num);
+        self.persisted_trie_updates.retain(|_, (block_num, _)| *block_num >= finalized_num);
```

Comment:
this can even be, right @Rjected ?
```suggestion
        self.persisted_trie_updates.retain(|_, (block_num, _)| *block_num > finalized_num);
```

Reply:
yes, `>` should work